src/battlearena/SearchPlayers.js

```diff
@@ -0,0 +1,93 @@
+//@flow strict
+
+import * as React from 'react';
+
+type Props = {|
+	playerList: Array<string>,
+|};
+
+type State = {|
+	searchName: string,
+	searchList: Array<string>
+|};
+
+// Search Players Component. Takes an array of all players.
+//
+// Prop Names:
+// playerList (Array of all users)
+//
+// State Names:
+// searchName (the keyword the user is searching with)
+// searchList (list of all users that match searchName)
+// 
+// EXAMPLE PROP USAGE = <SearchPlayers playerList={ ["jeffery", "jeff", "john", "illeatbutt", "childeatbigtime", "childconsumer"] }/>
+//
+// Needs to be updated to have a challenge player button.
+class SearchPlayers extends React.Component<Props, State> {
+
+	constructor(props: Props) {
+		super(props);
+
+		this.state = {
+			searchName: "",
+			searchList: []
+		}
+	}
+
+	// When typing in the searchbar, update searchName.
+	handleChange = (e: string) => {
+		this.setState({
+			searchName: e
+		});
+	}
+
+	// When a user clicks search, handleSearch will activate.
+	handleSearch = () => {

```

Comment:
Don't forget return types here.

I think it looks like: handleSearch = (): void => { ...
but I never do it like this, so I'm not sure